clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -14025,6 +14028,19 @@ Value *CodeGenFunction::EmitX86CpuInit() {
   return Builder.CreateCall(Func);
 }
 
+Value *CodeGenFunction::EmitAArch64CpuSupports(const CallExpr *E) {
+  const Expr *ArgExpr = E->getArg(0)->IgnoreParenCasts();
+  StringRef ArgStr = cast<StringLiteral>(ArgExpr)->getString();
```

Comment:
Does prior machinery in sema for the builtin forbid e.g:

```
const char *feats = "rng+flagm";
__builtin_cpu_supports(feats);
```

If not, this cast may fail.

Reply:
oh, yes. it does.